File: lc_soln/0271_encode_and_decode_strings/solution.cs
public class Solution {

    public string Encode(IList<string> strs) {
        var sb = new StringBuilder();
        for (int i = 0; i < strs.Count; ++i) {
            if (i != 0) {
                sb.Append(',');
            }
            sb.Append(strs[i].Length);
        }
        sb.Append('#');
        foreach (string s in strs) {
            sb.Append(s);
        }
        return sb.ToString();
    }

    public List<string> Decode(string s) {
        string[] parts = s.Split('#', 2);
        int[] lengths = Array.ConvertAll(parts[0].Split(','), int.Parse);
        List<string> result = [];
        int start = 0;
        foreach (int length in lengths) {
            result.Add(parts[1].Substring(start, length));
            start += length;
        }
        return result;
   }
}

File: lc_soln/0271_encode_and_decode_strings/solution.py
from typing import List


class Solution:
    def encode(self, strs: List[str]) -> str:
        lengths = []
        for s in strs:
            lengths.append(str(len(s)))
        return f"{','.join(lengths)}#{''.join(strs)}"

    def decode(self, s: str) -> List[str]:
        lengths, rest = s.split("#", 1)
        if not lengths:
            return []
        result = []
        start = 0
        for length in map(int, lengths.split(",")):
            result.append(rest[start : start + length])
            start += length
        return result


def main(): ...


if __name__ == "__main__":
    main()
